Remotion/Core/Extensions/Logging/Log4Net/Log4NetLogger.cs

```diff
@@ -0,0 +1,139 @@
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Globalization;
+using log4net.Core;
+using log4net.Util;
+using MicrosoftLogging = Microsoft.Extensions.Logging;
+using RemotionLogging = Remotion.Logging;
+
+namespace Remotion.Logging.Log4Net;
+
+public class Log4NetLogger : MicrosoftLogging.ILogger
+{
+  private readonly log4net.Core.ILogger _logger;
+
+  public Log4NetLogger (string categoryName)
+  {
+    _logger = log4net.LogManager.GetLogger(categoryName).Logger;
+  }
+
+  public static Level Convert (MicrosoftLogging.LogLevel logLevel)
+  {
+    switch (logLevel)
+    {
+      case MicrosoftLogging.LogLevel.Debug:
+        return Level.Debug;
+      case MicrosoftLogging.LogLevel.Information:
+        return Level.Info;
+      case MicrosoftLogging.LogLevel.Warning:
+        return Level.Warn;
+      case MicrosoftLogging.LogLevel.Error:
+        return Level.Error;
+      case MicrosoftLogging.LogLevel.Critical:
+        return Level.Critical;
+      default:
+        throw new ArgumentException(string.Format("LogLevel does not support value {0}.", logLevel), "logLevel");
+    }
+  }
+
+  public static Level Convert (RemotionLogging.LogLevel logLevel)
+  {
+    switch (logLevel)
+    {
+      case RemotionLogging.LogLevel.Debug:
+        return Level.Debug;
+      case RemotionLogging.LogLevel.Info:
+        return Level.Info;
+      case RemotionLogging.LogLevel.Warn:
+        return Level.Warn;
+      case RemotionLogging.LogLevel.Error:
+        return Level.Error;
+      case RemotionLogging.LogLevel.Fatal:
+        return Level.Fatal;
+      default:
+        throw new ArgumentException(string.Format("LogLevel does not support value {0}.", logLevel), "logLevel");
+    }
+  }
+
+  public void Log<TState> (MicrosoftLogging.LogLevel logLevel, MicrosoftLogging.EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
+  {
+    var level = Convert(logLevel);
+    if (_logger.IsEnabledFor(level))
+      _logger.Log(CreateLoggingEvent(level, eventId.Id, formatter(state, exception), exception));
+  }
+
+  public bool IsEnabled (MicrosoftLogging.LogLevel logLevel)
+  {
+    return _logger.IsEnabledFor(Convert(logLevel));
+  }
+
+  public IDisposable? BeginScope<TState> (TState state)
+      where TState : notnull
+  {
+    return null;
+  }
+
+  private LoggingEvent CreateLoggingEvent (Level level, int? eventID, object? message, Exception? exceptionObject)
+  {
+    var loggingEvent = new LoggingEvent(typeof(RemotionLogging.Log4NetLog), null, _logger.Name, level, message, exceptionObject);
```

Comment:
We don't want to use `Log4NetLog` here, as it would create a dependency on something that we want to remove.